core/arch/arm/tee/arch_svc.c

```diff
@@ -235,7 +236,7 @@ bool user_ta_handle_svc(struct thread_svc_regs *regs)
 	if (scn > TEE_SCN_MAX)
 		scf = (syscall_t)syscall_not_supported;
 	else
-		scf = tee_svc_syscall_table[scn].fn;
+		scf = tee_svc_syscall_table[SPECULATION_SAFE_VALUE(scn)].fn;
```

Comment:
what about this? if isb/msb here is too expensive.
```
	syscall_t *min = tee_svc_syscall_table;
	syscall_t *max = min + TEE_SCN_MAX;
	syscall_t *ptr = min + scn;

	scf = load_no_speculate_fail(ptr, min, max, syscall_not_supported);

Reply:
well, actually this is not compliant with `load_no_speculate_fail()`.
Sorry, forget my comment.

Reply:
An alternative could be:
```c
	scf = (syscall_t)syscall_not_supported;
	for (size_t idx = 0; idx < TEE_SCN_MAX; idx++)
		if (idx == scn)
			scf = tee_svc_syscall_table[SPECULATION_SAFE_VALUE(idx)].fn;

Reply:
I don't think isb/dsb is _that_ expensive.